src/deck/drivers/src/aideck.c

```diff
@@ -283,9 +328,133 @@ void cpxInitRoute(const CPXTarget_t source, const CPXTarget_t destination, const
     route->lastPacket = true;
 }
 
-static void aideckInit(DeckInfo *info)
+static CPXPacket_t bootPacket;
+
+#define FLASH_BUFFER_SIZE 64
+static uint8_t flashBuffer[FLASH_BUFFER_SIZE];
+static int flashBufferIndex = 0;
+
+static bool gap8DeckFlasherWrite(const uint32_t memAddr, const uint8_t writeLen, const uint8_t *buffer, const DeckMemDef_t* memDef) {
+
+  cpxInitRoute(CPX_T_STM32, CPX_T_GAP8, CPX_F_BOOTLOADER, &bootPacket.route);
+
+  if (memAddr == 0) {
+    GAP8BlCmdPacket_t* gap8BlPacket = (GAP8BlCmdPacket_t*)bootPacket.data;
+
+    gap8BlPacket->cmd = GAP8_BL_CMD_START_WRITE;
+    gap8BlPacket->startAddress = 0x40000;
+    gap8BlPacket->writeSize = *(memDef->newFwSizeP);
+    bootPacket.dataLength = sizeof(GAP8BlCmdPacket_t);
+    cpxSendPacketBlocking(&bootPacket);
+  }
+
+  // The GAP8 can only flash data in multiples of 4 bytes,
+  // buffering will guard against this and also speed things up.
+  // The full binary that will be flashed is multiple of 4.
+
+  uint32_t sizeLeftToBufferFull = sizeof(flashBuffer) - flashBufferIndex;
+  uint32_t sizeAbleToBuffer = sizeLeftToBufferFull < writeLen ? sizeLeftToBufferFull : writeLen;
+  uint32_t lastAddressToWrite = memAddr + sizeAbleToBuffer;
+
+  memcpy(&flashBuffer[flashBufferIndex], buffer, sizeAbleToBuffer);
+  flashBufferIndex += sizeAbleToBuffer;
+
+  if (flashBufferIndex == sizeof(flashBuffer) || lastAddressToWrite == *(memDef->newFwSizeP)) {
+    memcpy(&bootPacket.data, flashBuffer, flashBufferIndex);
+    bootPacket.dataLength = flashBufferIndex;
+
+    cpxSendPacketBlocking(&bootPacket);
+
+    flashBufferIndex = 0;
+    int sizeLeftToBuffer = writeLen - sizeLeftToBufferFull;
+    if (sizeLeftToBuffer > 0) {
+      memcpy(&flashBuffer[flashBufferIndex], &buffer[sizeLeftToBufferFull], sizeLeftToBuffer);
+      flashBufferIndex += sizeLeftToBuffer;
+    }
+  }
+
+  return true;
+}
+
+
+static bool isGap8InBootloaderMode = false;
+
+static void resetGap8ToBootloader() {
+  cpxInitRoute(CPX_T_STM32, CPX_T_ESP32, CPX_F_SYSTEM, &bootPacket.route);
+
+  ESP32SysPacket_t* esp32SysPacket = (ESP32SysPacket_t*)bootPacket.data;
+
+  esp32SysPacket->cmd = ESP32_SYS_CMD_RESET_GAP8;
+  bootPacket.dataLength = sizeof(ESP32SysPacket_t);
+
+  cpxSendPacketBlocking(&bootPacket);
+  // This should be handled on RX on CPX instead
+  vTaskDelay(100);
```

Comment:
100 ms should be more than enough to restart the GAP8 in bootloader mode, but as we get more use-cases maybe we should something that more resembles the router in the ESP and move this up a level so we can also easily do RX on different functions.